File: pitcher-service/src/integrationTest/java/io/pitchcast/pitcherservice/web/pitchercontroller/PitcherController_getIT.java
package io.pitchcast.pitcherservice.web.pitchercontroller;

import com.fasterxml.jackson.databind.ObjectMapper;
import io.pitchcast.pitcherservice.domain.Pitcher;
import io.pitchcast.pitcherservice.service.PitcherService;
import io.pitchcast.pitcherservice.web.PitcherController;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
import org.springframework.boot.test.mock.mockito.MockBean;
import org.springframework.http.MediaType;
import org.springframework.test.context.junit4.SpringRunner;
import org.springframework.test.web.servlet.MockMvc;
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
import org.springframework.test.web.servlet.result.MockMvcResultMatchers;

import java.util.Arrays;

import static io.github.benas.randombeans.api.EnhancedRandom.random;
import static org.mockito.BDDMockito.given;
import static org.mockito.Mockito.times;
import static org.mockito.Mockito.verify;

@RunWith(SpringRunner.class)
@WebMvcTest(controllers = PitcherController.class)
public class PitcherController_getIT {

    @Autowired
    private MockMvc mockMvc;

    @Autowired
    private ObjectMapper objectMapper;

    @MockBean
    private PitcherService pitcherService;

    @Test
    public void shouldReturn200IfPitcher() throws Exception {

        given(this.pitcherService.getAllPitcher()).willReturn(Arrays.asList(
                random(Pitcher.class),
                random(Pitcher.class)
        ));

        mockMvc.perform(
                MockMvcRequestBuilders.get("/pitcher/get")
                .contentType(MediaType.APPLICATION_JSON)
        ).andExpect(MockMvcResultMatchers.status().is(200));

        verify(pitcherService, times(1)).getAllPitcher();
    }

}

File: pitcher-service/src/integrationTest/kotlin/io/pitchcast/pitcherservice/web/PitcherController_getIT.kt
package io.pitchcast.pitcherservice.web

import io.github.benas.randombeans.api.EnhancedRandom.random
import io.pitchcast.pitcherservice.domain.Pitcher
import io.pitchcast.pitcherservice.service.PitcherService
import org.junit.jupiter.api.Test
import org.mockito.BDDMockito.given
import org.mockito.Mockito.times
import org.mockito.Mockito.verify
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.mock.mockito.MockBean
import org.springframework.http.MediaType
import org.springframework.test.web.servlet.MockMvc
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
import org.springframework.test.web.servlet.result.MockMvcResultMatchers
import testing.WebMvcSliceTest

@WebMvcSliceTest(forController = [PitcherController::class])
class PitcherController_getIT {

    @Autowired lateinit var mockMvc: MockMvc
    @MockBean lateinit var pitcherService: PitcherService

    @Test fun `should return 200 if pitcher`() {

        given<List<Pitcher>>(this.pitcherService.allPitcher).willReturn(arrayListOf<Pitcher>(
                random<Pitcher>(Pitcher::class.java),
                random<Pitcher>(Pitcher::class.java)
        ))

        mockMvc.perform(
                MockMvcRequestBuilders.get("/pitcher/")
                        .contentType(MediaType.APPLICATION_JSON)
        ).andExpect(MockMvcResultMatchers.status().`is`(200))

        verify<PitcherService>(pitcherService, times(1)).getAllPitcher()
    }

}